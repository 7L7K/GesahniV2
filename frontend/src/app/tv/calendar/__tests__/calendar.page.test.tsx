import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Page from '../page';

jest.mock('@/lib/api', () => ({
  apiFetch: jest.fn(async (_: string, __: any) => ({
    json: async () => ({ items: [{ time: '10:00', title: 'Doctor' }] })
  }))
}));

describe('TV Calendar page', () => {
  it('renders next items and matches snapshot', async () => {
    const { container } = render(<Page />);
    await waitFor(() => expect(screen.getByText(/Doctor/)).toBeInTheDocument());
    expect(container.firstChild).toMatchSnapshot();
  });
});
