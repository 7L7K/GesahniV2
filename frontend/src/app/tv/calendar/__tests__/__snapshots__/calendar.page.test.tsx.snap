// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TV Calendar page renders next items and matches snapshot 1`] = `
<main
  class="min-h-screen bg-black text-white p-8 max-w-4xl mx-auto"
>
  <h1
    class="text-6xl font-bold mb-8"
  >
    Today
  </h1>
  <ul
    class="space-y-4"
  >
    <li
      class="bg-zinc-800 rounded-3xl p-6 text-4xl flex items-center gap-6"
    >
      <div
        class="w-48 text-right opacity-90"
      >
        10:00
      </div>
      <div
        class="flex-1 font-semibold"
      >
        Doctor
      </div>
    </li>
  </ul>
  <div
    class="mt-10"
  >
    <button
      class="bg-white text-black px-10 py-6 rounded-3xl text-4xl"
    >
      Mute reminders today?
    </button>
  </div>
</main>
`;
